modules/stochastic_tools/include/utils/PolynomialQuadrature.h

```diff
@@ -43,6 +43,9 @@ class Polynomial
   virtual void quadrature(const unsigned int order,
                           std::vector<Real> & points,
                           std::vector<Real> & weights) const = 0;
+  virtual void clenshawQuadrature(const unsigned int order,
```

Comment:
Why do you need this? This goes against the "polymorphic" nature of the base class.

Also, should these be const variables?

Reply:
Not sure what you mean by "polymorphic". This function is meant to be analogous to the `quadrature` function above it. Technically, the `quadrature` function should be `gaussQuadrature`. But Gauss quadrature is pretty ubiquitous, it's the quadrature that everyone knows and loves. Clenshaw-Curtis quadrature is really only useful for polynomial chaos with sparse grids.

The vectors aren't const because they are the output.

Reply:
http://www.cplusplus.com/doc/tutorial/polymorphism/